### ISSUE TITLE
How can I run E2E testing on Gitlab CI?

### ISSUE THREAD

**Issue by weihongyu12** [vuejs/vue-cli#4723]
### Version
4.0.3

### Reproduction link
[https://github.com/weihongyu12/vue-ci](https://github.com/weihongyu12/vue-ci)

### Environment info
```
 System:
    OS: Windows 10
    CPU: (4) x64 Intel(R) Core(TM) i5-4200M CPU @ 2.50GHz
  Binaries:
    Node: 10.16.3 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.19.1 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.9.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: 44.18362.387.0
  npmPackages:
    @vue/babel-helper-vue-jsx-merge-props:  1.0.0
    @vue/babel-plugin-transform-vue-jsx:  1.0.0
    @vue/babel-preset-app:  4.0.3 (3.12.0)
    @vue/babel-preset-jsx:  1.1.1
    @vue/babel-sugar-functional-vue:  1.0.0
    @vue/babel-sugar-inject-h:  1.0.0
    @vue/babel-sugar-v-model:  1.1.1
    @vue/babel-sugar-v-on:  1.1.0
    @vue/cli-overlay:  4.0.3
    @vue/cli-plugin-babel: ^4.0.3 => 4.0.3
    @vue/cli-plugin-e2e-nightwatch: ^4.0.3 => 4.0.3
    @vue/cli-plugin-eslint: ^4.0.3 => 4.0.3
    @vue/cli-plugin-pwa: ^4.0.3 => 4.0.3
    @vue/cli-plugin-router: ^4.0.3 => 4.0.3
    @vue/cli-plugin-unit-mocha: ^4.0.3 => 4.0.3
    @vue/cli-plugin-vuex: ^4.0.3 => 4.0.3
    @vue/cli-service: ^4.0.3 => 4.0.3
    @vue/cli-shared-utils:  4.0.3
    @vue/component-compiler-utils:  3.0.0
    @vue/eslint-config-airbnb: ^4.0.1 => 4.0.1
    @vue/preload-webpack-plugin: ^1.1.1 => 1.1.1
    @vue/test-utils: 1.0.0-beta.29 => 1.0.0-beta.29
    @vue/web-component-wrapper:  1.2.0
    eslint-plugin-vue: ^5.2.3 => 5.2.3
    typescript:  3.6.4
    vue: ^2.6.10 => 2.6.10
    vue-docgen-api:  3.25.0
    vue-eslint-parser:  5.0.0 (2.0.3)
    vue-hot-reload-api:  2.3.4
    vue-inbrowser-compiler:  3.23.0
    vue-loader:  15.7.1
    vue-router: ^3.1.3 => 3.1.3
    vue-server-renderer:  2.6.10
    vue-style-loader:  4.1.2
    vue-styleguidist: ^3.25.0 => 3.25.0
    vue-template-compiler: ^2.6.10 => 2.6.10
    vue-template-es2015-compiler:  1.9.1
    vuepress: ^1.2.0 => 1.2.0
    vuepress-html-webpack-plugin:  3.2.0
    vuepress-plugin-container:  2.0.2
    vuepress-plugin-smooth-scroll:  0.0.3
    vuex: ^3.1.1 => 3.1.1
  npmGlobalPackages:
    @vue/cli: Not Found
```

### Steps to reproduce
Locally used vue-cli generated code, E2E test configuration selected nightwatch.js, `selenium-server` installed and selenium for E2E testing, only simple configuration in `.gitlab-ci.yml`.

```yml
e2e test:chrome:
  image: node:10
  services:
    - selenium/standalone-chrome
  stage: test
  script:
    - yarn install --frozen-lockfile
    - yarn test:e2e --env chrome --headless
```

On the Gitlab Runner server, Docker is used and `selenium/standalone-chrome` is installed.

But when I push the code to the repository, I get an error when I run the E2E test.

### What is expected?
Does anyone know the specific solution? If there is a solution, please let me know, thank you!

### What is actually happening?
The following error occurred while running CI.

```
$ vue-cli-service test:e2e --use-selenium --env chrome --headless
 INFO  Starting development server...
 WARNING  Compiled with 2 warnings1:26:16 AM

 warning  

asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets: 
  js/chunk-vendors.403f0b0b.js (273 KiB)

 warning  

entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  app (281 KiB)
      js/chunk-vendors.403f0b0b.js
      css/app.24df6853.css
      js/app.4d6792b7.js



  App running at:
  - Local:   http://localhost:8080/ 

  It seems you are running Vue CLI inside a container.
  Access the dev server via http://localhost:<your container's external mapped port>/

  App is served in production mode.
  Note this is for preview or E2E testing only.

 INFO  Running end-to-end tests in headless mode...
   Error: ChromeDriver process exited with code: 127
   /builds/root/vue-template/node_modules/chromedriver/lib/chromedriver/chromedriver: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory
  
       at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
 /builds/root/vue-template/node_modules/chromedriver/lib/chromedriver/chromedriver: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory
 
 ERROR  Error: Command failed: /builds/root/vue-template/node_modules/nightwatch/bin/nightwatch --env chrome --config /builds/root/vue-template/node_modules/@vue/cli-plugin-e2e-nightwatch/nightwatch.config.js
Error: Command failed: /builds/root/vue-template/node_modules/nightwatch/bin/nightwatch --env chrome --config /builds/root/vue-template/node_modules/@vue/cli-plugin-e2e-nightwatch/nightwatch.config.js
    at makeError (/builds/root/vue-template/node_modules/execa/index.js:174:9)
    at Promise.all.then.arr (/builds/root/vue-template/node_modules/execa/index.js:278:16)
    at process._tickCallback (internal/process/next_tick.js:68:7)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: Job failed: exit code 1
```

---
There are very few documents on CI and E2E testing, I think it should be perfected.

**Comment by LinusBorg** [comment]
It seems that chromedriver fails run in your docker container. since that is a dependency of nightwatch (not Vue CLI), you would probably be better of asking in their community for known issues with chromedriver in docker contexts.

Also, if you google the error message you find that this error also happened with chrome on linux for some people in the past:

https://www.linuxquestions.org/questions/slackware-14/google-chrome-error-loading-libnss3-so-811621/

In short, this is not a problem we will address in the context of Vue CLI since it's about chromedriver & linux (& docker), not the abstraction that Vue CLI provider over nightwatch.

**Comment by weihongyu12** [comment]
Thanks this is really not caused by Vue CLI, but because Chrome is not running properly in the container

I have solved the problem now. The correct approach is to install Chorme or Firefox before running E2E testing.

```yml
e2e test:
  image: node:lts
  stage: test
  before_script:
    - apt-get update
    # Install Chrome
    - apt-get install -yq libnss3 unzip openjdk-8-jre-headless xvfb libxi6 libgconf-2-4
    - wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
    - curl -sS -o - https://dl.google.com/linux/linux_signing_key.pub | apt-key add -
    - apt install -y ./google-chrome-stable_current_amd64.deb
    # Install Firefox
    - apt-get install -y firefox-esr
  script:
    - yarn install --frozen-lockfile
    - yarn test:e2e
```
